### PR TITLE
feat(wren-ai-service): sql-expansion

### PR DESCRIPTION
**Pull request by cyyeh**
- refactoring
  - remove global variables
  - restructuring pipelines
  - fix test command in Justfile
- sql-expansion

### CODE REVIEW

**Comment by cyyeh** [comment]
image
there is issue in openai api currently, and we can safely merge this PR if no other issues

File: wren-ai-service/src/pipelines/generation/sql_regeneration.py [hunk 1]
```diff
@@ -109,20 +155,24 @@ async def sql_regeneration_post_process(
...
-        self.sql_regeneration_preprocesser = SQLRegenerationRreprocesser()
-        self.sql_regeneration_prompt_builder = PromptBuilder(
-            template=sql_regeneration_user_prompt_template
-        )
-        self.sql_regeneration_generator = llm_provider.get_generator(
-            system_prompt=sql_regeneration_system_prompt
-        )
-        self.sql_regeneration_post_processor = GenerationPostProcessor(engine=engine)
+        self._components = {
+            "sql_regeneration_preprocesser": SQLRegenerationRreprocesser(),
```
Comment (paopa) [inline wren-ai-service/src/pipelines/generation/sql_regeneration.py:165]: here seems a typo.

```suggestion
            "sql_regeneration_preprocessor": SQLRegenerationPreprocessor(),
```

File: wren-ai-service/src/pipelines/generation/sql_regeneration.py [hunk 2]
```diff
@@ -109,20 +155,24 @@ async def sql_regeneration_post_process(
...
+            "sql_regeneration_prompt_builder": PromptBuilder(
+                template=sql_regeneration_user_prompt_template
+            ),
+            "sql_regeneration_generator": llm_provider.get_generator(
+                system_prompt=sql_regeneration_system_prompt
+            ),
+            "sql_regeneration_post_processor": SQLBreakdownGenerationPostProcessor(
```
Comment (paopa) [inline wren-ai-service/src/pipelines/generation/sql_regeneration.py:172]: I think we could consider removing the prefix `sql_regeneration` to be more concise.

File: wren-ai-service/src/pipelines/retrieval/historical_question.py [hunk 3]
```diff
@@ -143,14 +143,17 @@ def __init__(
...
     ) -> None:
-        self._store = store_provider.get_store(dataset_name="view_questions")
-        self._embedder = embedder_provider.get_text_embedder()
-        self._retriever = store_provider.get_retriever(
-            document_store=self._store,
-        )
-        self._score_filter = ScoreFilter()
-        # todo: add a llm filter to filter out low scoring document
-        self._output_formatter = OutputFormatter()
+        _store = store_provider.get_store(dataset_name="view_questions")
```
Comment (paopa) [inline wren-ai-service/src/pipelines/retrieval/historical_question.py:146]: nitpick: I do believe there is you just forgot removing the underscore. 🤣

File: wren-ai-service/src/providers/engine/wren.py [hunk 4]
```diff
@@ -96,7 +96,7 @@ async def execute_sql(
             if response.status == 204:
                 return True, None, None
 
-            res = await response.json()
+            res = await response.text()
```
Comment (paopa) [inline wren-ai-service/src/providers/engine/wren.py:99]: curiously, why did you change to use text method to get the response?

File: wren-ai-service/src/pipelines/generation/sql_regeneration.py (hunk 1 above)
Comment (cyyeh) [reply wren-ai-service/src/pipelines/generation/sql_regeneration.py]: fixed

File: wren-ai-service/src/pipelines/generation/sql_regeneration.py (hunk 2 above)
Comment (cyyeh) [reply wren-ai-service/src/pipelines/generation/sql_regeneration.py]: fixed

File: wren-ai-service/src/pipelines/retrieval/historical_question.py (hunk 3 above)
Comment (cyyeh) [reply wren-ai-service/src/pipelines/retrieval/historical_question.py]: fixed

File: wren-ai-service/tests/pytest/services/test_ask.py [hunk 5]
```diff
@@ -76,6 +72,14 @@ def indexing_service():
     )
 
 
+@pytest.fixture
+def service_metadata():
+    return {
+        "models_metadata": {},
+        "service_version": "",
+    }
```
Comment (paopa) [inline wren-ai-service/tests/pytest/services/test_ask.py:80]: nitpick: it seems the same as the default which the function `trace_metadata` can't got from kwargs, I think giving a mock value is better to identify for testing and the default

File: wren-ai-service/tests/pytest/services/test_ask_details.py [hunk 6]
```diff
@@ -17,17 +17,27 @@ def ask_details_service():
...
 
 
+@pytest.fixture
+def service_metadata():
+    return {
+        "models_metadata": {},
+        "service_version": "",
+    }
+
+
```
Comment (paopa) [inline wren-ai-service/tests/pytest/services/test_ask_details.py:35]: nitpick: same as the comment before, gently mention again.

**Review by paopa — APPROVED** [review]
there has a minor comment for the test case. others LGTM. thanks for the hard work!
